### PR TITLE
[run_nn] Printing instantaneous values for each batch

### PR DESCRIPTION
**Pull request by akashmjn**
Printing out error, loss during processing of a minibatch that's useful for debugging and inspection. 
Also max_len of sequence values are useful while balancing GPU memory usage and tuning batch size and related parameters. 

The gradient clipping part that's currently commented out also logs useful stats about the gradients, helpful while fixing the norm value to clip at.

### CODE REVIEW

**Comment by mravanelli** [comment]
I tested and it works. The only problem I see is with the "len" field since if you use a feed-forward network instead of RNN is value is 0 and than can be misleading for many people. Moreover, I think that the performance on each batch is very noisy and can be misleading. Let me think a little bit more about it...

**Comment by akashmjn** [comment]
Aah right, didn't think about the feedforward case for "len", probably optional in that case. 

However having access to instantaneous batch values I would say is pretty essential to have as a user as opposed to waiting till the end of processing a chunk to discover nans or gradient explosions, especially during initial coarse hyperparam tuning. Useful for a future tensorboardX integration as well that has plenty of smoothing options.

**Comment by mravanelli** [comment]
Yes,  in the next version I would like to think something like a
tensorboard X integration...

Mirco

**Comment by mravanelli** [comment]
The current version contains the print. Thank you!
